docs_src/design/adr/core/0021-Device-Profile-Changes.md

```diff
@@ -0,0 +1,66 @@
+
+### Summary of Device Profile Rules
+These rules will be implemented in core metadata on device profile API calls.
+
+- A device profile can be added anytime
+- Device resources or device commands can be added to a device profile anytime
+- Attributes can be added to a device profile anytime
+- A device profile can be removed or modified when the device profile is not associated to a device or provision watcher
+    - this includes modifying any field (except identifiers like names and ids)
+    - this includes changes to the array of device resources, device commands
+    - this includes changes to attributes (of device resources)
+    - even when a device profile is associated to a device or provision watcher, fields of the device profile or device resource can be modified when the field change will not affect the behavior of the system (example: the description field has no effect on the system).
+- A device profile cannot be removed when it is associated to a device or provision watcher.
+- A device profile can be removed or modified even when associated to an event or reading.  However, configuration options (see New Configuration Settings below) are available to block the change or removal of a device profile for any reason.
+    - the rationale behind the new configuraton settings was specifically to protect the event/reading association to device profiles.  Events and readings are generally considered short lived (ephemeral) objects and already contain the necessary device profile information that are needed by the system during their short life without having to refer to and keep the device profile.  But if an adopter wants to make sure the device profile is unmodified and still exists for any event/readings association (or for any reason), then the new config setting will block device profile changes or removals.
+    - see note below in Consequences that a new Units property must be added to the Reading object in order to support this rule and the need for all relevant profile data to be in the reading.
+
+### Ancillary Rules associated to Device Profiles
+- Identifying or “key” fields for device profiles, device resources, etc. cannot be modified and can never be null.
+- A device profile can begin life “empty” - meaning that it has no device resources or device commands.
+
+### New APIs
+
+The following APIs would be added to the metadata REST service in order to meet the design specified above.
+
+- Add Profile General Property PATCH API (allow to modify profile field except name and id)
```

Comment:
I'm not sure what "allow to modify profile field except name and id" means? Why don't we just say "(used to modify the profile's Manufacturer, Model, and Label fields)"?

Reply:
changed accordingly